### PR TITLE
RTL FlowDirection fix for Windows

### PR DESCRIPTION
**Pull request by buyaa-n**
### Description of Change

There is inconsistency on windows when `FlowDirection` is set to `RighToLeft`:
Windows inconsistency
View for `Start` and `End` values of `HorizontalOptions` are flipped for windows. We have [manual flip handling](https://github.com/dotnet/maui/blob/main/src/Core/src/Layouts/LayoutExtensions.cs#L83) for RTL layout for controls, but the default `FlowDirection` handling for layouts is causing another flip on windows and therefore RTL flipping is being applied twice  on windows. The suggested fix was overwrite default `FlowDirection` handling on windows

### Issues Fixed

This is related to https://github.com/dotnet/maui/issues/4742, fixing inconsistency found during investigating that issue, the reported issue is not a bug, it was by design, but the team considering to update that design, so i am not closing that issue with this PR. The image after fix:

After fix

### CODE REVIEW

File: src/Controls/samples/Controls.Sample/Pages/Layouts/StackLayoutPage.xaml [hunk 1]
```diff
@@ -30,6 +30,28 @@
...
+            <Label Text="Horizontal alignment FlowDirection = RightToLeft"/>
+            <StackLayout WidthRequest="400" FlowDirection="RightToLeft" BackgroundColor="AliceBlue">
+                <Button HeightRequest="50" Text="Button start" HorizontalOptions="Start" />
+                <Label HeightRequest="50" Text="Label start" HorizontalOptions="Start" />
+                <Button HeightRequest="50" Text="Button end" HorizontalOptions="End" />
+                <Label HeightRequest="50" Text="Label end" HorizontalOptions="End" />
+                <Button HeightRequest="50" Text="Button Fill" HorizontalOptions="Fill" />
+                <Label HeightRequest="50"  Text="Label Fill" HorizontalOptions="Fill" />
+                <Button HeightRequest="50" WidthRequest="100" Text="Button fill" HorizontalOptions="Fill" />
+                <Label HeightRequest="50"  WidthRequest="100" Text="Label fill" HorizontalOptions="Fill" />
```
Comment (buyaa-n) [inline src/Controls/samples/Controls.Sample/Pages/Layouts/StackLayoutPage.xaml:53]: I don't feel like this layout flip is testable with unit test, so added a sample view, but let me know if i am wrong

File: src/Core/tests/DeviceTests/Stubs/LayoutStub.cs [hunk 2]
```diff
@@ -84,5 +84,11 @@ public Size CrossPlatformArrange(Rect bounds)
 		public bool ClipsToBounds { get; set; }
 
 		public IView this[int index] { get => _children[index]; set => _children[index] = value; }
+
+		public override FlowDirection FlowDirection
+		{
+			get => FlowDirection.LeftToRight;
+			set { }
+		}
```
Comment (buyaa-n) [inline src/Core/tests/DeviceTests/Stubs/LayoutStub.cs:92]: https://github.com/dotnet/maui/blob/93092a0486bb48f53fd275d8aff7fd50644a822f/src/Core/tests/DeviceTests/Handlers/HandlerTestBaseOfT.Tests.cs#L22-L33
The above test is failing on windows, i do not see a better fix that this, open for suggestions

Comment (buyaa-n) [inline src/Core/tests/DeviceTests/Stubs/LayoutStub.cs:92]: https://github.com/dotnet/maui/blob/93092a0486bb48f53fd275d8aff7fd50644a822f/src/Core/tests/DeviceTests/Handlers/HandlerTestBaseOfT.Tests.cs#L22-L33
The above test is failing on windows, i do not see a better fix that this, open for suggestions

File: src/Core/tests/DeviceTests/Stubs/LayoutStub2.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Maui.DeviceTests.Stubs
+{
+	public class LayoutStub2 : LayoutStub
+	{
+		public override FlowDirection FlowDirection { get; set; } = FlowDirection.LeftToRight;
```
Comment (buyaa-n) [inline src/Core/tests/DeviceTests/Stubs/LayoutStub2.cs:11]: For the new test added, i need a `LayoutStub`  that returns the real value set for `FlowDirection`

Comment (buyaa-n) [inline src/Core/tests/DeviceTests/Stubs/LayoutStub2.cs:11]: For the new test added, i need a `LayoutStub`  that returns the real value set for `FlowDirection`

File: src/Controls/samples/Controls.Sample/Pages/Layouts/StackLayoutPage.xaml (hunk 1 above)
Comment (buyaa-n) [reply src/Controls/samples/Controls.Sample/Pages/Layouts/StackLayoutPage.xaml]: Removed this and added unit test

Comment (buyaa-n) [reply src/Controls/samples/Controls.Sample/Pages/Layouts/StackLayoutPage.xaml]: Removed this and added unit test

**Review by hartez — APPROVED** [review]

**Pull request merged**